### PR TITLE
Chore: Catalog Widget - Replace Block Escaping with Escaper

### LINKED ISSUE #37107: [Issue] Chore: Catalog Widget - Replace Block Escaping with Escaper
This issue is automatically created based on existing pull request: magento/magento2#37082: Chore: Catalog Widget - Replace Block Escaping with Escaper

---------
### Description (*)
Refactors the `Magento_CatalogWidget` module to replace `$block` escaping functions with `$escaper` escaping functions.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by pykettk**
### Description (*)
Refactors the `Magento_CatalogWidget` module to replace `$block` escaping functions with `$escaper` escaping functions.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#37107: Chore: Catalog Widget - Replace Block Escaping with Escaper

### CODE REVIEW

**Comment by pykettk** [comment]
@magento create issue

**Review by andrewbess — APPROVED** [review]
Hello @pykettk 
Thank you for your contribution here.

**Comment by engcom-Lima** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**

**Manual testing scenario:**

- In code base verify $block replaces with $escaper in the `app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml`

and `app/code/Magento/CatalogWidget/view/frontend/templates/product/widget/content/grid.phtml`

**Before:** :heavy_multiplication_x: 

Screenshot 2023-03-15 at 5 18 30 PM

**After:** :heavy_check_mark:  
Screenshot 2023-03-15 at 4 00 35 PM

**Additionally tested:**

1. Create, Edit and deleted the widget.
2. Verified the created widget functionality in the frontend.
3. Verified the admin grid of widget.

No additional manual testing is required as a part of regression cycle as this PR is related to the refactoring of the code base.

As builds are failed. Hence, moving this PR to **Extended testing**

**Comment by engcom-Delta** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Static Tests, Unit Tests

**Comment by engcom-Delta** [comment]
@magento run Static Tests

**Comment by engcom-Charlie** [comment]
Hi @pykettk,

Thank you for the contribution!

As your branch is forked from https://github.com/mage-os/mageos-magento2.git, we can not able to work on it. Can you please have a look at them else we have to create a new PR from branch forked from https://github.com/magento/magento2.git and close this one.

Thank you!

**Comment by pykettk** [comment]
@engcom-Charlie What changes would you like me to make?

**Comment by engcom-Charlie** [comment]
@magento run Static Tests

**Comment by engcom-Lima** [comment]
Hi @pykettk ,

> @engcom-Charlie What changes would you like me to make?

In order to move ahead to **merge in progress** all builds should be green. As multiple builds are failed our team have to work on it but as your PR is forked from [mage-os](https://github.com/mage-os/mageos-magento2.git) we can't proceed further on it.
Please check this [comment](https://github.com/magento/magento2/pull/37082#issuecomment-1476251851) and please act accordingly. 

Thanks

**Comment by engcom-Charlie** [comment]
Hi @pykettk,

In general, when PR is showing build failures, we need to work on it. It will get move to Merge only when the build is successful. For this PR, we engcom team, not able to work on your PR as your branch is forked from https://github.com/mage-os/mageos-magento2.git 

As mentioned in [comment](https://github.com/magento/magento2/pull/37082#issuecomment-1476251851), either you please have a look at the build failures or please raise a new PR from branch forked from https://github.com/magento/magento2.git and close this one.

Thank you!

**Comment by pykettk** [comment]
@engcom-Charlie I'd really like to look at the failing checks however the report is unavailable - see below.
image

**Comment by pykettk** [comment]
@magento run Static Tests

File: app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml [hunk 1]
```diff
@@ -3,17 +3,23 @@
...
+
+/** @var Escaper $escaper */
+/** @var Conditions $block */
+/** @var SecureHtmlRenderer $secureRenderer */
 
 $element = $block->getElement();
-$fieldId = $element->getHtmlContainerId() ? ' id="' . $block->escapeHtmlAttr($element->getHtmlContainerId()) . '"' : '';
-$fieldClass = 'field admin__field field-' . $block->escapeHtmlAttr($element->getId()) . ' '
-    . $block->escapeHtmlAttr($element->getCssClass());
+$fieldId = $element->getHtmlContainerId() ? ' id="' . $escaper->escapeHtmlAttr($element->getHtmlContainerId()) . '"' : '';
```
Comment (fredden) [inline app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml:17]: `phpcs` says this line is too long (122 characters, limit is 120)

File: app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml [hunk 2]
```diff
@@ -31,8 +37,8 @@ $fieldAttributes = $fieldId . ' class="' . $fieldClass . '" '
         "Magento_Rule/rules",
         "prototype"
     ], function(VarienRulesForm){
-        window.{$block->escapeJs($block->getHtmlId())} = new VarienRulesForm('{$block->escapeJs($block->getHtmlId())}',
-         '{$block->escapeUrl($block->getNewChildUrl())}');
+        window.{$escaper->escapeJs($block->getHtmlId())} = new VarienRulesForm('{$escaper->escapeJs($block->getHtmlId())}',
```
Comment (fredden) [inline app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml:40]: `phpcs` says this line is too long (123 characters, limit is 120)

**Review by fredden — COMMENTED** [review]
I've manually run `vendor/bin/phpcs --standard=Magento2Framework -sp --basepath=$PWD --colors app/code/Magento/CatalogWidget/view/frontend/templates/product/widget/content/grid.phtml app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml` and this is what it is complaining about:

```
FILE: app/code/Magento/CatalogWidget/view/adminhtml/templates/product/widget/conditions.phtml
--------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES
--------------------------------------------------------------------------------------------------------
 17 | WARNING | Line exceeds 120 characters; contains 122 characters
    |         | (Generic.Files.LineLength.TooLong)
 40 | WARNING | Line exceeds 120 characters; contains 123 characters
    |         | (Generic.Files.LineLength.TooLong)
--------------------------------------------------------------------------------------------------------
```

I suspect this is why the automated Static Tests are failing. There were no complains from `vendor/bin/phpmd` which is also run as part of the automated Static Tests.

**Comment by fredden** [comment]
> I've manually run `vendor/bin/phpcs --standard=Magento2Framework ...`

Note that I have https://github.com/magento/magento-coding-standard/pull/417 applied locally, so when I say `Magento2Framework`, I mean `Magento2` and `Magento2Framework`.

**Comment by pykettk** [comment]
Thanks for the input, @fredden , this is very helpful! I'll look to get these resolved now 🙂

**Pull request closed without merging**